### ISSUE TITLE
Homekit will not turn send command if device is already in that state

### ISSUE THREAD

**Issue by Astroman** [Colorado4Wheeler/HomeKit-Bridge#119]
### Expected Behavior

After upgrading to iOS 12 (may have nothing to do with iOS 12), Siri now will not send an on or off command to a device if she thinks the device is already in that state. This, for example, affects all of my X10 light switches after I turn them off manually. It affects many other devices like Morningstar lock controller, RF controlled rolling shades, etc. I don't see any attempt to control them in the Indigo log or anywhere else except that Siri responds as if she controlled the device. I don't think there is any Siri or Homekit log available so I don't know how to check further. It is possible, but unlikely, that Homekit or Homekit Bridge is filtering the command but I don't know how to tell for sure if this is the case.

### Current Behavior

No command is issued as far as I can tell. I would expect the command to be sent as it was before a couple of weeks ago. 

### Steps to Reproduce (for bugs)

Turn light on with Siri so that status becomes "On". Turn light off manually with X10 switch so that Siri still thinks it is on. Ask Siri to turn the light on. She responds that she is turning it on but no command is issued.

### Versions

- **Plugin**: `YourVersion`
1.0.2

### Configuration

```json
 HomeKit Bridge                  {
        "platforms": [
                {
                        "protocol": "http", 
                        "name": "HomeKit Bridge Server", 
                        "listenPort": "8446", 
                        "platform": "Indigo2", 
                        "host": "127.0.0.1", 
                        "debug": false, 
                        "serverId": xxxxxxxxxx, 
                        "port": "8558"
                }
        ], 
        "bridge": {
                "username": "XX:XX:XX:XX:XX", 
                "name": "Homekit Bridge", 
                "pin": "XXX-XX-XXX", 
                "port": "51828"
        }, 
        "accessories": [], 
        "description": "HomeKit configuration generated by HomeKit Bridge on 2018-09-23 21:21:46.355000 for device Homekit Bridge"
}
```

Not relevant AFAIK
### Homebridge Log / Command Output

```
No log output is recorded when this happens
```
### Device Simulation Report

```
##### DEVICE SIMULATION DATA #####
	
	Service : 
		alias : Family Room Light
		model : 
		subModel : 
		indigoType : indigo.DimmerDevice
		pluginType : Built-In
		type : Lightbulb
		desc : Lightbulb
		objId : 1525081080
		serverId : 1669194149
		invertOnState : False
		convertFahrenheit : False
		required : (List)
			On : True
		optional : (List)
			Hue : 0.0
			Saturation : 0.0
			ColorTemperature : 140
			Name
			Brightness : 100
		native : True
		requiresPlugin : (List)
		actions : (List)
			Action : (HomeKitAction)
				Characteristic : On
				When : equal
				Value : False (bool)
				Value2 : 0 (bool)
				Command : device.turnOff
				Arguments : [1525081080]
				monitors : {1525081080: 'attr_onState'}
			Action : (HomeKitAction)
				Characteristic : On
				When : equal
				Value : True (bool)
				Value2 : 0 (bool)
				Command : device.turnOn
				Arguments : [1525081080]
				monitors : {1525081080: 'attr_onState'}
			Action : (HomeKitAction)
				Characteristic : Brightness
				When : between
				Value : 0 (int)
				Value2 : 100 (int)
				Command : dimmer.setBrightness
				Arguments : [1525081080, '=value=']
				monitors : {1525081080: 'attr_brightness'}
		loadOptional : False
		characterDict : (Dict)
			On : True
			Brightness : 100
	

	[/code]
```

**Comment by Astroman** [comment]
After hours of discussion with Apple and getting a response from their engineering group, I've gotten nowhere with this. They say I need to talk with the device manufacturer which obviously wouldn't help. I strongly suspect that Homekit changed its behavior but I can't rule out that Homekit-Bridge or homebridge-indigo isn't filtering out the commands when the device is already reportedly in the desired state.

Does anyone know how I could tell if Homekit is actually sending the command to my network?

**Comment by Colorado4Wheeler** [comment]
This all only started after upgrading to iOS 12?  in iOS 11 it was fine?  As far as I know nothing significant has changed in HomeKit between the two versions and I'm running this on iOS 12 without any issues at all.

> No log output is recorded when this happens

That's not possible, there is a log and there is no way it is blank.  If the log is blank then the device in question is not part of HomeKit Bridge.

**Comment by Astroman** [comment]
Thanks for the reply. I assume by now you’ve seen all the forum messages about this. This apparently started when I recently converted from Homebridge Buddy and I didn’t notice it until I updated to ios 12 a short time later.  Nor did I realize the plugin is intentionally ignoring these type of requests. I finally got the messages about the command being ignored to show up (the log wasn’t blank; just those messages weren’t showing up until I executed the advanced action).  I implemented a fix suggested by Webdeck and my devices are responding again as expected. Will let you know if any unexpected problems crop up. Thanks!
